pkg/util/kubernetes/apiserver/util.go

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build kubeapiserver
+
+package apiserver
+
+import (
+	"context"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/client-go/tools/cache"
+)
+
+// syncTimeout can be used to wait for the kubernetes client-go cache to sync.
+var syncTimeout = config.Datadog.GetDuration("cache_sync_timeout") * time.Second
+
+// SyncInformers should be called after the instanciation of new informers.
+// It's blocking until the informers are synced or the timeout exceeded.
+func SyncInformers(informers []cache.SharedInformer) error {
+	var g errgroup.Group
+	for _, inf := range informers {
+		g.Go(func() error {
+			ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(syncTimeout))
+			defer cancel()
+			if !cache.WaitForCacheSync(ctx.Done(), inf.HasSynced) {
```

Comment:
should we consider `WaitForNamedCacheSync` so we know which one fails? https://godoc.org/k8s.io/client-go/tools/cache#WaitForNamedCacheSync

Reply:
not sure if the informer logs appear with the cluster agent logs 🤔  https://github.com/kubernetes/client-go/blob/master/tools/cache/shared_informer.go#L239-L249

Reply:
Ah yes, the good old custom logger. If they don't, maybe we should propagate the informer name in order to add it to the log line. Otherwise it's not actionable. We already have some names in the controller catalog, not sure if it's a canonical source but it's something: https://github.com/DataDog/datadog-agent/pull/5291/files#diff-905ee21984317f631a5c9d6743500518R27